testing/regress/ecl/despray.ecl

```diff
@@ -0,0 +1,223 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2016 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+//nothor
+
+import Std.File AS FileServices;
+
+unsigned VERBOSE := 0;
+
+Layout_Person := RECORD
+  UNSIGNED1 PersonID;
+  STRING15  FirstName;
+  STRING25  LastName;
+END;
+
+allPeople := DATASET([ {1,'Fred','Smith'},
+                       {2,'Joe','Blow'},
+                       {3,'Jane','Smith'}],Layout_Person);
+
+//  Outputs  ---
+output(allPeople, , '~persons', OVERWRITE);
+
+import * from lib_fileservices;
+
+SrcAddrIp := '.';
+SrcAddrLocalhost := 'localhost';
+File := 'persons';
+SourceFile := '~::' + File;
+
+ClusterName := 'mythor';
+DestFile1 := '/var/lib/HPCCSystems/mydropzone/' + File;
+
+rec := RECORD
+  string sourceFile;
+  string destFile;
+  string ip;
+  boolean allowOverwrite;
+  string result;
+  string msg;
+end;
+
+rec t(rec l) := TRANSFORM
+  SELF.sourceFile := l.sourceFile;
+  SELF.msg := FileServices.fDespray(l.sourceFile
+                                         ,l.ip
+                                         ,destinationPath := l.destFile
+                                         ,ALLOWOVERWRITE := l.allowOverwrite
+                                         );
+  SELF.result := 'Pass';
+  SELF.ip := l.ip;
+  SELF.allowOverwrite := l.allowOverwrite;
+  SELF.destFile := l.destFile;
+end;
+
+// Target path validation chekcing
+
+dst0 := NOFOLD(DATASET([{SourceFile, DestFile1, SrcAddrIp, True, '', ''}], rec));
+p0 := PROJECT(NOFOLD(dst0), t(LEFT));
+c0 := CATCH(p0, ONFAIL(TRANSFORM(rec,
+                                 SELF.sourceFile := SourceFile,
+                                 SELF.destFile := DestFile1,
+                                 SELF.ip := '',
+                                 SELF.allowOverwrite := True,
+                                 SELF.result := 'Fail',
+                                 SELF.msg := FAILMESSAGE
+                                )));
+#if (VERBOSE = 1)
+    output(c0);
+#else
+    output(c0, {result});
+#end
+
+
+dst1 := NOFOLD(DATASET([{SourceFile, DestFile1, SrcAddrLocalhost, True, '', ''}], rec));
```

Comment:
It would be helpful to have a comment above each test to indicate what it was testing and the expected result.